diff -ur a/engine/Makefile b/engine/Makefile
--- a/engine/Makefile	2019-10-10 02:24:21.000000000 +0800
+++ b/engine/Makefile	2021-11-13 17:44:38.851501469 +0800
@@ -669,7 +669,7 @@
 $(TARGET) : $(OBJS) $(RES)
 	@echo
 	@echo Linking $(TARGET_PLATFORM) Port: $(TARGET)...
-	@$(CC) $(CFLAGS) -o $(TARGET) $(OBJS) $(TARGET_RESOURCE) $(LIBS)
+	@$(CC) $(CFLAGS) -o $(TARGET) $(OBJS) $(TARGET_RESOURCE) -ldl $(LIBS)
 $(TARGET_FINAL) : $(TARGET)
 	@echo Stripping $(TARGET_PLATFORM) Port: $(TARGET_FINAL)...
 	@$(STRIP)
