--- a/arch/arm64/boot/dts/amlogic/meson-g12b.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b.dtsi
@@ -48,6 +48,7 @@
 			compatible = "arm,cortex-a53";
 			reg = <0x0 0x0>;
 			enable-method = "psci";
+      capacity-dmips-mhz = <592>;
 			next-level-cache = <&l2>;
 		};
 
@@ -56,6 +57,7 @@
 			compatible = "arm,cortex-a53";
 			reg = <0x0 0x1>;
 			enable-method = "psci";
+      capacity-dmips-mhz = <592>;
 			next-level-cache = <&l2>;
 		};
 
@@ -64,6 +66,7 @@
 			compatible = "arm,cortex-a73";
 			reg = <0x0 0x100>;
 			enable-method = "psci";
+      capacity-dmips-mhz = <1024>;
 			next-level-cache = <&l2>;
 		};
 
@@ -72,6 +75,7 @@
 			compatible = "arm,cortex-a73";
 			reg = <0x0 0x101>;
 			enable-method = "psci";
+      capacity-dmips-mhz = <1024>;
 			next-level-cache = <&l2>;
 		};
 
@@ -80,6 +84,7 @@
 			compatible = "arm,cortex-a73";
 			reg = <0x0 0x102>;
 			enable-method = "psci";
+      capacity-dmips-mhz = <1024>;
 			next-level-cache = <&l2>;
 		};
 
@@ -88,6 +93,7 @@
 			compatible = "arm,cortex-a73";
 			reg = <0x0 0x103>;
 			enable-method = "psci";
+      capacity-dmips-mhz = <1024>;
 			next-level-cache = <&l2>;
 		};
 

