From 9691c0b6631ff5323bc5400d1b127ee2141ab244 Mon Sep 17 00:00:00 2001
From: 5schatten <supervisedthinking@gmail.com>
Date: Thu, 30 May 2019 10:55:11 +0200
Subject: [PATCH] [Makefile] fix cross compile

---
 Makefile | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index e26994478..191edd3d1 100644
--- a/Makefile
+++ b/Makefile
@@ -21,16 +21,7 @@ HAVE_CLANG    := 0
 
 TARGET_NAME   := beetledc
 
-ifeq ($(HAVE_CLANG),1)
-	CXX      = ${CC_PREFIX}clang++
-	CC       = ${CC_PREFIX}clang
-	SHARED   := -fuse-ld=lld
-else
-	CXX      = ${CC_PREFIX}g++
-	CC       = ${CC_PREFIX}gcc
-	SHARED   :=
-endif
-CC_AS    = ${CC_PREFIX}as
+MFLAGS   :=
 
 GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
 ifneq ($(GIT_VERSION)," unknown")
@@ -848,7 +839,7 @@ LIBS     += -lm
 PREFIX        ?= /usr/local
 
 ifneq (,$(findstring arm, $(ARCH)))
-	CC_AS    = ${CC_PREFIX}gcc #The ngen_arm.S must be compiled with gcc, not as
+	CC_AS    = ${CC_PREFIX}${CC} #The ngen_arm.S must be compiled with gcc, not as
 	ASFLAGS  += $(CFLAGS)
 endif
 
