From 573cd4cdd56f09ff6a27a552093b63906b919ed9 Mon Sep 17 00:00:00 2001
From: Cristi Mitrana <31816814+cmitu@users.noreply.github.com>
Date: Wed, 12 Jun 2019 13:36:48 +0300
Subject: [PATCH] Fix 'Last Played' collection crash.

---
 es-app/src/CollectionSystemManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/es-app/src/CollectionSystemManager.cpp b/es-app/src/CollectionSystemManager.cpp
index cb5e2481d..0ccdd9ee1 100644
--- a/es-app/src/CollectionSystemManager.cpp
+++ b/es-app/src/CollectionSystemManager.cpp
@@ -292,7 +292,7 @@ void CollectionSystemManager::updateCollectionSystem(FileData* file, CollectionS
 void CollectionSystemManager::trimCollectionCount(FileData* rootFolder, int limit)
 {
 	SystemData* curSys = rootFolder->getSystem();
-	while (rootFolder->getChildren().size() > limit)
+	while (rootFolder->getChildrenListToDisplay().size() > limit)
 	{
 		CollectionFileData* gameToRemove = (CollectionFileData*)rootFolder->getChildrenListToDisplay().back();
 		ViewController::get()->getGameListView(curSys).get()->remove(gameToRemove, false);
