From bfa83a58b8393c1798b432d09c61b90b6f7ec1be Mon Sep 17 00:00:00 2001
From: 5schatten <supervisedthinking@gmail.com>
Date: Tue, 27 Aug 2019 21:39:35 +0200
Subject: [PATCH] CMakeList.txt: add option to disable libCEC / enabled by
 default

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9542da48c..23e5ef305 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,7 @@ cmake_minimum_required(VERSION 2.8)
 option(GLES "Set to ON if targeting Embedded OpenGL" ${GLES})
 option(GL "Set to ON if targeting Desktop OpenGL" ${GL})
 option(RPI "Set to ON to enable the Raspberry PI video player (omxplayer)" ${RPI})
+option(USE_LIBCEC "Set to ON to enable libCEC" ON)
 
 project(emulationstation-all)
 
@@ -59,7 +60,11 @@ find_package(SDL2 REQUIRED)
 find_package(CURL REQUIRED)
 find_package(VLC REQUIRED)
 find_package(RapidJSON REQUIRED)
-find_package(libCEC)
+
+#add libCEC support
+if(USE_LIBCEC)
+    find_package(libCEC REQUIRED)
+endif()
 
 #add ALSA for Linux
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
