From 672a0be46ac929d2af8eac7666a1ca518bca9ee3 Mon Sep 17 00:00:00 2001
From: 5schatten <supervisedthinking@gmail.com>
Date: Thu, 11 Apr 2019 00:06:05 +0200
Subject: [PATCH] CMakeList: added symlinks for libGLES_CM.so.1 &
 libGLESv1_CM.so.1

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad43cdf..18db8a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,6 +58,8 @@ if(DEFINED MALI_FILE)
     install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libMali.so)
     install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libEGL.so)
     install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libEGL.so.1)
+    install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libGLES_CM.so.1)
+    install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libGLESv1_CM.so.1)
     install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libGLESv2.so)
     install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libGLESv2.so.2)
     install_symlink(libmali.so ${CMAKE_INSTALL_LIBDIR}/libgbm.so)
